src/Processors/QueryPlan/Optimizations/useDataParallelAggregation.cpp

```diff
@@ -202,7 +206,7 @@ size_t tryAggregatePartitionsIndependently(QueryPlan::Node * node, QueryPlan::No
         return 0;
 
     if (!reading->willOutputEachPartitionThroughSeparatePort()
-        && isPartitionKeySuitsGroupByKey(*reading, expression_step->getExpression(), *aggregating_step))
+        && isPartitionKeySuitsGroupByKey(*reading, expression_step->getExpression()->clone(), *aggregating_step))
```

Comment:
Not sure if `clone()` is necessary here. I don't see it's changing inside `isPartitionKeySuitsGroupByKey`

Reply:
`removeUnusedActions` modifies actions dag